File: nodejs/src/api/handlers/forms/getAllForms/getAllFormsHandler.js
'use strict';

const AWS = require('aws-sdk')
const db = new AWS.DynamoDB.DocumentClient({})

const tableName = process.env.FORMS_TABLE_NAME

const response = (statusCode, message) => {
    return {
        statusCode: statusCode,
        body: JSON.stringify(message)
    }
}

const getAllForms = async () => {
    const params = {
      TableName: tableName,
      IndexName: 'ListFormsTitles',
      KeyConditionExpression: 'GSI1PK = :v',
      ExpressionAttributeValues: {
        ':v': 'FORMS'
      }
    }
  
    try {
      const formsSearch = await db.query(params).promise()
      const forms = formsSearch.Items
      return response(200, forms)
    } catch (err) {
      return response(err.statusCode, err)
    }
}

module.exports = {
  getAllForms
}

File: python/src/api/handlers/forms/getAllForms/getAllFormsHandler.py
try:
    import unzip_requirements
except ImportError:
    pass

import boto3
import os
import simplejson as json

dynamodb_client = boto3.resource('dynamodb')
TABLE_NAME = os.environ['FORMS_TABLE_NAME']
DYNAMO_FORMS_ID = os.environ['FORMS_ID']

def create_response(code, message):
  return {
    'statusCode': code,
    'headers': {'Content-Type': 'application/json'},
    'body': json.dumps(message)
  }

def getAllForms(event, context):


    params = {
        'IndexName': 'ListFormsTitles',
        'KeyConditionExpression': 'GSI1PK = :v',
        'ExpressionAttributeValues': {
            ':v': 'FORMS'
        }
    }

    try:
        table = dynamodb_client.Table(TABLE_NAME)
        response = table.query(**params)
    except Exception as ex:
        return create_response(500, str(ex))

    print('response: ', response)

    if response['ResponseMetadata']['HTTPStatusCode'] != 200:
      return create_response(500, 'Error getting the forms')

    forms = response['Items']

    return create_response(200, forms)

